src/Workspaces/MSBuildTest/Utilities/DotNetSdkMSBuildInstalled.cs

```diff
@@ -22,18 +24,27 @@ internal partial class DotNetSdkMSBuildInstalled : ExecutionCondition
 
         static DotNetSdkMSBuildInstalled()
         {
-            if (!TryGetSDKVersion(out var version, out var rootPath))
+            if (TryGetSDKVersion(out var version, out var rootPath))
             {
-                return;
-            }
+                VersionString = version;
 
-            if (TryGetSdkPath(rootPath, version, out var sdkPath, out var msbuildPath))
+                if (TryGetSdkPath(rootPath, version, out var sdkPath, out var msbuildPath))
+                {
+                    MSBuildLocator.RegisterMSBuildPath(msbuildPath);
+                    SdkPath = sdkPath;
+                }
+            }
+            else
             {
-                MSBuildLocator.RegisterMSBuildPath(msbuildPath);
-                SdkPath = sdkPath;
+                // We don't have a global.json at all, but if we can find exactly one SDK version, we'll use that. This supports running in Helix.
+                var instance = MSBuildLocator.QueryVisualStudioInstances(new VisualStudioInstanceQueryOptions { DiscoveryTypes = DiscoveryType.DotNetSdk }).SingleOrDefault();
+
+                if (instance != null)
```

Comment:
should this throw if no instance is ever located?

Reply:
If it throws or if it does this, the test will skip either way.